runtime/hsa-runtime/core/inc/amd_aie_aql_queue.h

```diff
@@ -131,6 +159,45 @@ class AieAqlQueue : public core::Queue,
   /// @brief Base of the queue's ring buffer storage.
   void *ring_buf_ = nullptr;
 
+  static hsa_status_t SubmitCmd(
```

Comment:
It seems like all these functions should belong to the XDNA driver and be used by the soft queue processor, but should not be part of the queue.

Reply:
Moved the work submission logic from the queue to the driver in 530dfee, let me know what you think!